### ISSUE TITLE
Remove const hack in prefer_const_constructors

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#1555]
https://github.com/dart-lang/linter/blob/c42d2383682b37454138b2845c9c4f37d55ef262/lib/src/rules/prefer_const_constructors.dart#L89-L99

Trying to use `context.canBeConst(node)` I face an issue related to generics. The following line is now flag but shouldn't:

```dart
void gimmeJ<T>() => new J<T>(); // OK
```

/cc @scheglov

**Comment by a14n** [comment]
Fixed by #1595
